tools/autograd/derivatives.yaml

```diff
@@ -1267,6 +1267,10 @@
   self: grad * std::sqrt(2 * M_PI) * (result.square() / 2).exp()
   result: auto_element_wise
 
+- name: special_log_ndtr(Tensor self) -> Tensor
+  self: grad / std::sqrt(2 * M_PI) * (result + self.pow(2) / 2).neg().exp()
```

Comment:
this gradient definition uses 'result', thus may incur large memory usage. 
Alternatively, one can recompute with at::ndtr(self), but since ndtr is a computation heavy op, I chose to trade memory for speed.

Reply:
Also, do I need to support complex number?

Reply:
In case the user wants to reduce memory usage, they can use checkpointing :)

Reply:
I have seen a few cases where user ended up writing their own backward impl with autograd.Function. :)

Reply:
Avoiding recomputation is definitely the default choice we make here. So sounds good.
You don't have to support complex derivatives for such functions. If you do, you can add it to the list at https://github.com/pytorch/pytorch/blob/ffd9608963b0a6bbbed7c9fee33553628304a630/tools/autograd/gen_variable_type.py#L87 to make sure they are properly tested.

Reply:
we don't support complex numbers for functions in special module, so don't worry about it for this PR